cranelift/codegen/src/isa/x64/inst/emit.rs

```diff
@@ -1049,45 +1051,56 @@ pub(crate) fn emit(inst: &Inst, sink: &mut MachBuffer<Inst>) {
             );
         }
 
-        Inst::XMM_RM_R {
+        Inst::XMM_MOV_RM_R {
             op,
             src: srcE,
             dst: reg_g,
         } => {
             let rex = RexFlags::clear_w();
+            let (prefix, opcode) = match op {
+                SseOpcode::Movaps => (LegacyPrefix::None, 0x0F28),
+                SseOpcode::Movd => (LegacyPrefix::_66, 0x0F6E),
+                SseOpcode::Movsd => (LegacyPrefix::_F2, 0x0F10),
+                SseOpcode::Movss => (LegacyPrefix::_F3, 0x0F10),
+                _ => unimplemented!("Opcode {:?} not implemented", op),
+            };
 
-            let opcode = match op {
-                SseOpcode::Addss => 0x0F58,
-                SseOpcode::Subss => 0x0F5C,
-                SseOpcode::Mulss => 0x0F59,
-                SseOpcode::Divss => 0x0F5E,
-                SseOpcode::Sqrtss => 0x0F51,
-                _ => unimplemented!("XMM_RM_R opcode"),
+            match srcE {
+                RegMem::Reg { reg: regE } => {
```

Comment:
nit: can you rename `regE` to `reg_e` please?

Reply:
Note ... there will be an inconsistency here w.r.t. to some other cases in this match. Will probably want to revisit those and make sure it is all consistent. For this patch will not touch those other cases as I think initially there may have been a subtlety behind choosing regE vs reg_e for naming that we should make sure is consistent everywhere.